TimoCloud-Universal/src/main/java/cloud/timo/TimoCloud/core/sockets/CoreStringHandler.java

```diff
@@ -44,337 +30,47 @@ public void handleMessage(Message message, String originalMessage, Channel chann
-                channel.attr(CoreRSAHandshakeHandler.HANDSHAKE_PERFORMED_ATTRIBUTE_KEY).set(true);
-                return;
-            }
-            case PROXY_HANDSHAKE: {
-                if (proxy == null) {
-                    closeChannel(channel);
-                    return;
-                }
-                if (! (address.equals(proxy.getBase().getAddress()) || address.equals(proxy.getBase().getPublicAddress()))) {
-                    TimoCloudCore.getInstance().severe("Proxy connected with different InetAddress than its base. Refusing connection.");
-                    return;
-                }
-                if (! channel.attr(CoreRSAHandshakeHandler.RSA_KEY_ATTRIBUTE_KEY).get().equals(proxy.getPublicKey())) {
-                    TimoCloudCore.getInstance().severe(String.format("Proxy %s connected with wrong public key. Please report this.", proxy.getName()));
-                    return;
-                }
-                TimoCloudCore.getInstance().getSocketServerHandler().setCommunicatable(channel, proxy);
-                proxy.onConnect(channel);
-                proxy.onHandshakeSuccess();
-                channel.attr(CoreRSAHandshakeHandler.HANDSHAKE_PERFORMED_ATTRIBUTE_KEY).set(true);
-                return;
-            }
-            case BASE_HANDSHAKE: {
-                if (TimoCloudCore.getInstance().getInstanceManager().isBaseConnected(baseName)) {
-                    TimoCloudCore.getInstance().severe("Error while base handshake: A base with the name '" + baseName + "' is already conencted.");
-                    return;
-                }
-                InetAddress publicAddress = address;
-                try {
-                    publicAddress = InetAddress.getByName((String) message.get("publicAddress"));
-                } catch (Exception e) {
-                    TimoCloudCore.getInstance().severe("Unable to resolve public ip address '" + message.get("publicAddress") + "' for base " + baseName + ". Please make sure the base's hostname is configured correctly in your operating system.");
-                }
-                PublicKey publicKey = channel.attr(CoreRSAHandshakeHandler.RSA_KEY_ATTRIBUTE_KEY).get();
 
-                if (! TimoCloudCore.getInstance().getCorePublicKeyManager().redeemBaseKeyIfPermitted(publicKey)) {
-                    channel.close();
-                    return;
-                }
+        boolean handshake = false;
 
-                Base base = TimoCloudCore.getInstance().getInstanceManager().getBaseByPublicKey(publicKey);
-                if (base == null) { // First connection
-                    base = TimoCloudCore.getInstance().getInstanceManager().createBase(publicKey);
+        try {
+            for (MessageHandler messageHandler : getMessageHandlers(type)) {
+                if (messageHandler.getMessageType().toString().contains("HANDSHAKE")) {
+                    handshake = true;
+                    messageHandler.execute(message, channel);
```

Comment:
We need some kind of exception handling so that the execution does not stop if one of the handlers throws an exception